File: src/api/company_announcement.ts
// src/api/company_announcement.ts
import http from '@/utils/http'

export interface Announcement {
  announcement_id: string
  company_id: string
  title: string
  content: string
  target_major?: string
  target_degree?: number
  headcount?: number
  deadline?: string
  status: number
  published_at?: string
  created_at: string
}

export interface AnnouncementCreate {
  title: string
  content: string
  target_major?: string
  target_degree?: number
  headcount?: number
  deadline?: string
  status?: number
}

export interface AnnouncementQuery {
  status?: number
  year?: number
  page?: number
  page_size?: number
}

export const getAnnouncements = (params?: AnnouncementQuery) =>
  http.get<{ list: Announcement[]; total: number; page: number; page_size: number }>({
    url: '/api/v1/company/announcements',
    params
  })

export const getAnnouncement = (id: string) =>
  http.get<Announcement>({ url: `/api/v1/company/announcements/${id}` })

export const createAnnouncement = (data: AnnouncementCreate) =>
  http.post<Announcement>({ url: '/api/v1/company/announcements', data })

export const updateAnnouncement = (id: string, data: Partial<AnnouncementCreate>) =>
  http.put<Announcement>({ url: `/api/v1/company/announcements/${id}`, data })

export const deleteAnnouncement = (id: string) =>
  http.del({ url: `/api/v1/company/announcements/${id}` })

File: backend/app/schemas/company_announcement.py
from pydantic import BaseModel, Field
from typing import Optional
from datetime import date, datetime


class AnnouncementCreate(BaseModel):
    title: str = Field(..., max_length=200)
    content: str
    target_major: Optional[str] = Field(None, max_length=200)
    target_degree: Optional[int] = None
    headcount: Optional[int] = None
    deadline: Optional[date] = None
    status: int = 1


class AnnouncementUpdate(BaseModel):
    title: Optional[str] = Field(None, max_length=200)
    content: Optional[str] = None
    target_major: Optional[str] = Field(None, max_length=200)
    target_degree: Optional[int] = None
    headcount: Optional[int] = None
    deadline: Optional[date] = None
    status: Optional[int] = None


class AnnouncementOut(BaseModel):
    announcement_id: str
    company_id: str
    title: str
    content: str
    target_major: Optional[str]
    target_degree: Optional[int]
    headcount: Optional[int]
    deadline: Optional[date]
    status: int
    published_at: Optional[datetime]
    created_at: datetime

    model_config = {"from_attributes": True}